### PR TITLE
Getting started 1.4

### PR DESCRIPTION
**Pull request by JoeyZwicker**

### CODE REVIEW

File: doc/getting_started/beginner_tutorial.rst [hunk 1]
```diff
@@ -2,13 +2,10 @@ Beginner Tutorial
...
-Analyzing Log Lines from a Fruit Stand
---------------------------------------
+Image processing with OpenCV
+----------------------------
 
-In this guide you're going to create a Pachyderm pipeline to process
-transaction logs from a fruit stand. We'll use two standard unix tools, ``grep``
-and ``awk`` to do our processing. Thanks to Pachyderm's processing system we'll
-be able to run the pipeline in a distributed, streaming fashion. As new data is
+In this guide we're going to create a Pachyderm pipeline to do some simple edge detection on a few images. Thanks to Pachyderm's processing system, we'll be able to run the pipeline in a distributed, streaming fashion. As new data is
```
Comment (dwhitena) [inline doc/getting_started/beginner_tutorial.rst]: Maybe link edge detection to https://en.wikipedia.org/wiki/Edge_detection.  Not guaranteed everyone will know right off the bat depending on background.

File: doc/getting_started/beginner_tutorial.rst [hunk 2]
```diff
@@ -2,13 +2,10 @@ Beginner Tutorial
...
 added, the pipeline will automatically process it and materialize the results.
```
Comment (dwhitena) [inline doc/getting_started/beginner_tutorial.rst]: "materialize" can carry a bit of baggage.  Maybe just "output"?

File: doc/getting_started/beginner_tutorial.rst [hunk 3]
```diff
@@ -25,163 +22,245 @@ Create a Repo
 A ``repo`` is the highest level primitive in the Pachyderm file system (pfs). Like all primitives in pfs, it shares it's name with a primitive in Git and is designed to behave analogously. Generally, repos should be dedicated to a single source of data such as log messages from a particular service, a users table, or training data for an ML model. Repos are dirt cheap so don't be shy about making tons of them.
```
Comment (dwhitena) [inline doc/getting_started/beginner_tutorial.rst]: Do we want  talk a lot about PFS here?  Maybe "highest level data primitive in Pachyderm.  Like many things in Pachyderm, it shares it's name with primitives of Git..."

File: doc/getting_started/beginner_tutorial.rst [hunk 4]
```diff
@@ -25,163 +22,245 @@ Create a Repo
...
-Let's start by just adding a file to a new commit. We've provided a sample data file for you to use in our GitHub repo -- it's a list of purchases from a fruit stand.
+Let's start by just adding a file, in this case an image, to a new commit. We've provided some sample images for you that we host on Imgur. 
 
-We'll use the ``put-file`` command along with two flags, ``-c`` and ``-f``. ``-f`` can take either a local file or a URL, in our case, the sample data on GitHub.
+We'll use the ``put-file`` command along with two flags, ``-c`` and ``-f``. ``-f`` can take either a local file or a URL which it'll automatically scrape. In our case, we'll simply pass the URL.
 
- We also specificy the repo name "data" and the branch name "master".
+Unlike Git though, commits in Pachyderm must be explicitly started and finished as they can contain huge amounts of data and we don't want that much "dirty" data hanging around in an unpersisted state. The ``-c`` flag specifies that we want to start a new commit, add data, and finish the commit in a convenient one-liner.
+
+We also specify the repo name "images", the branch name "master", and what we want to name the file, "liberty,png".
```
Comment (dwhitena) [inline doc/getting_started/beginner_tutorial.rst]: "liberty.png"

File: doc/getting_started/beginner_tutorial.rst [hunk 5]
```diff
@@ -25,163 +22,245 @@ Create a Repo
...
+ $ pachctl get-file images master liberty.png | display
  ...
 
 Create a Pipeline
 ^^^^^^^^^^^^^^^^^
 
-Now that we've got some data in our repo, it's time to do something with it.
-``Pipelines`` are the core primitive for Pachyderm's processing system (pps) and
-they're specified with a JSON encoding. For this example, we've already created the pipeline for you and it can be found at `examples/fruit_stand/pipeline.json on Github <https://github.com/pachyderm/pachyderm/blob/master/doc/examples/fruit_stand/pipeline.json>`_. Please open a new tab to view the pipeline while we talk through it.
+Now that we've got some data in our repo, it's time to do something with it. ``Pipelines`` are the core primitive for Pachyderm's processing system (pps) and they're specified with a JSON encoding. For this example, we've already created the pipeline for you and you can find the `code on Github <https://github.com/pachyderm/pachyderm/blob/master/doc/examples/opencv>`_.
```
Comment (dwhitena) [inline doc/getting_started/beginner_tutorial.rst]: Again, just a question of whether we want to use PFS/PPS language or just Pachyderm pipeline.  I've been trying to about PFS/PPS.

File: doc/getting_started/beginner_tutorial.rst [hunk 6]
```diff
@@ -25,163 +22,245 @@ Create a Repo
...
-oranges and bananas. The second step sums these sales numbers into a final sales count.
+For now, we're going to create a single pipeline that takes in images and does some simple edge detection.
 
 .. code-block:: shell
 
- +----------+     +--------------+     +------------+
- |input data| --> |filter pipline| --> |sum pipeline|
- +----------+     +--------------+     +------------+
+ +----------+     +--------------+     +-------------+
+ |input data| --> |edge detection| --> | output data |
```
Comment (dwhitena) [inline doc/getting_started/beginner_tutorial.rst]: Maybe use the image:

opencv example

File: doc/getting_started/beginner_tutorial.rst [hunk 7]
```diff
@@ -25,163 +22,245 @@ Create a Repo
...
+images              634aa3728141442f8537a7b1bb6aefbf   <none>                             13 minutes ago     Less than a second   57.27 KiB
+
+# List the output commits
+$ pachctl list-commit edges
+REPO                ID                                 PARENT                             STARTED            DURATION             SIZE
+edges               8848e11056c04518a8d128b6939d9985   da51395708cb4812bc8695bb151b69e3   2 minutes ago      Less than a second   133.6 KiB
+edges               da51395708cb4812bc8695bb151b69e3   4dd2459531414d80936814b13b1a3442   2 minutes ago      Less than a second   59.38 KiB
+edges               4dd2459531414d80936814b13b1a3442   <none>                             5 minutes ago      Less than a second   22.22 KiB
+
+Exploring the File System (optional)
```
Comment (dwhitena) [inline doc/getting_started/beginner_tutorial.rst]: This is kind of cool, but I just wonder whether it's worth including here or maybe just including in the cookbook or somewhere.  There is the danger of them opening the directory structure and getting confused about how things are organized, duplicated, etc.  Not sure.  I trust your judgement either way.  Just provided an alternative point of view.

File: doc/getting_started/beginner_tutorial.rst [hunk 8]
```diff
@@ -200,36 +279,23 @@ And then mount it:
...
-- Add a new pipeline that does something interesting with the "sum" repo as an input.
-- Add your own data set and ``grep`` for different terms. This example can be generalized to generic word count.
-
-You can also start learning some of the more advanced topics to develop analysis in Pachyderm:
+We've now got Pachyderm running locally with data and a pipeline! If you want to keep playing with Pachyderm locally, you can  use what you've learned to build on or change this pipeline. You can also start learning some of the more advanced topics to develop analysis in Pachyderm:
 
-- :doc:`../deployment/deploying_on_the_cloud`
-- :doc:`../deployment/inputing_your_data` from other sources
-- :doc:`../deployment/custom_pipelines` using your own code
+- :doc:`../deployment/deploying_on_the_cloud` PLACEHOLDER
```
Comment (dwhitena) [inline doc/getting_started/beginner_tutorial.rst]: ../deployment/deploy_intro

File: doc/getting_started/beginner_tutorial.rst [hunk 9]
```diff
@@ -200,36 +279,23 @@ And then mount it:
...
+- :doc:`../deployment/inputing_your_data` from other sources PLACEHOLDER
```
Comment (dwhitena) [inline doc/getting_started/beginner_tutorial.rst]: ../fundamentals/getting_data_into_pachyderm

File: doc/getting_started/beginner_tutorial.rst [hunk 10]
```diff
@@ -200,36 +279,23 @@ And then mount it:
...
+- :doc:`../deployment/custom_pipelines` using your own code  PLACEHOLDER
```
Comment (dwhitena) [inline doc/getting_started/beginner_tutorial.rst]: ../fundamentals/creating_analysis_pipelines

File: doc/getting_started/getting_started.rst [hunk 11]
```diff
@@ -12,16 +12,16 @@ Welcome to the documentation portal for first time Pachyderm users! We've organi
 
 If you'd like to read about the technical concepts in Pachyderm before actually running it, check out our reference docs:
 
-* :doc:`../pachyderm_file_system`
-* :doc:`../pachyderm_pipeline_system`
-* :doc:`../pachctl/pachctl`
-* :doc:`../FAQ`
+* :doc:`../pachyderm_file_system` PLACEHOLDER
```
Comment (dwhitena) [inline doc/getting_started/getting_started.rst]: ../reference/pachyderm_file_system

File: doc/getting_started/getting_started.rst [hunk 12]
```diff
@@ -12,16 +12,16 @@ Welcome to the documentation portal for first time Pachyderm users! We've organi
...
+* :doc:`../pachyderm_pipeline_system` PLACEHOLDER
```
Comment (dwhitena) [inline doc/getting_started/getting_started.rst]: ../reference/pachyderm_pipeline_system

File: doc/getting_started/getting_started.rst [hunk 13]
```diff
@@ -12,16 +12,16 @@ Welcome to the documentation portal for first time Pachyderm users! We've organi
...
+* :doc:`../pachctl/pachctl` PLACEHOLDER
+* :doc:`../FAQ` PLACEHOLDER
 * `Use Cases <http://pachyderm.io/use_cases.html>`_
 
 ---
 
 
-If you've already got a Kubernetes cluster running or would rather use AWS, GCE or Azure, check out our :doc:`../deployment/deploying_on_the_cloud` documentation.
+If you've already got a Kubernetes cluster running or would rather use AWS, GCE or Azure, check out our :doc:`../deployment/deploying_on_the_cloud` documentation. PLACEHOLDER
```
Comment (dwhitena) [inline doc/getting_started/getting_started.rst]: ../deployment/deploy_intro

**Comment by dwhitena** [comment]
I think we need to add a note to the local deploy doc about how its best to start fresh every time with minikube (i.e., minikube + pachyderm doesn't respond well to restarts).  Just emphasize that minikube is good for testing, but it's not meant to be long running.

**Comment by JoeyZwicker** [comment]
I think in v1.4 it is idempotent and works pretty well -- certainly better than before where rethink coming up was often the problem. Probably better to be safe anyway and add it somewhere. Definitely will add in trouble shooting, but where else?

File: doc/getting_started/beginner_tutorial.rst (hunk 7 above)
Comment (JoeyZwicker) [inline doc/getting_started/beginner_tutorial.rst]: agreed. introduce fewer concepts.

**Comment by JoeyZwicker** [comment]
@dwhitena Ok I think this is ready for another review. I think all your changes have been made.

**Pull request merged**